### PR TITLE
Patch smoothing utility to handle short signals (for backfill)

### PR DESCRIPTION
**Pull request by dshemetov**
Mostly documentation fixes, confrm to pydocstyle, add missing documentation for the argument boundary_method.

But also I removed the len(signal) < window_length constraint I added at the last minute; I forgot I wrote boundary methods to handle that case a few months ago.

### CODE REVIEW

**Comment by dshemetov** [comment]
Should probably be merged before @chinandrew's refactors go live, so that backfill is handled properly.

File: _delphi_utils_python/tests/test_smooth.py [hunk 1]
```diff
@@ -167,11 +167,10 @@ def test_impute(self):
         with pytest.raises(ValueError):
             imputed_signal = smoother.savgol_impute(signal)
 
-        # test window_length > len(signal)
+        # test window_length > len(signal) and boundary_method="identity"
         signal = np.arange(20)
         smoother = Smoother(smoother_name="savgol", boundary_method="identity", window_length=30)
-        with pytest.raises(ValueError):
-            smoothed_signal = smoother.smooth(signal)
+        smoothed_signal = smoother.smooth(signal)
```
Comment (chinandrew) [inline _delphi_utils_python/tests/test_smooth.py:173]: should there be some assertion on the output of this? I know it used to be a value error and now it doesn't, so checking it returned right would further verify behavior

**Review by chinandrew — COMMENTED** [review]
one comment otherwise lgtm

**Comment by dshemetov** [comment]
Cool. You know, I'm running into issues where JHU tests are giving the smoother an array full of NANs. That makes me want to change the nan handling to be a bit more permissive (a full array of nans should pass through, an array left-padded with nans should start imputing only once the first value is seen, etc.), so I think I'll handle that in this PR too.

**Comment by chinandrew** [comment]
> Cool. You know, I'm running into issues where JHU tests are giving the smoother an array full of NANs. That makes me want to change the nan handling to be a bit more permissive (a full array of nans should pass through, an array left-padded with nans should start imputing only once the first value is seen, etc.), so I think I'll handle that in this PR too.

If it's independent of this change, I'd recommend making it another PR just to keep things clean.

**Comment by dshemetov** [comment]
Noted. I'll do that.

File: _delphi_utils_python/tests/test_smooth.py (hunk 1 above)
Comment (krivard) [reply _delphi_utils_python/tests/test_smooth.py:173]: +1, @dshemetov ?

Comment (dshemetov) [reply _delphi_utils_python/tests/test_smooth.py:173]: Good catch, added an assertion 👍

**Review by chinandrew — APPROVED** [review]
lgtm, only test failing is the one that #476 fixes.

**Review by krivard — APPROVED** [review]
lgtm

**Pull request merged**